### PR TITLE
feat: show the annotaiton sources only if we have more than 1 annotation source and disable filtering by annotation source

### PR DESCRIPTION
**Pull request by ikondrat**
## Motivation
When an annotation collection is created, an annotation source is automatically created alongside it. This causes confusion: the single annotation source that accompanies a newly created annotation collection is not meaningful to the user and clutters the UI unnecessarily.

The decision is to hide the annotation source selector in cases where the user has not intentionally run an evaluation (i.e. there is only one annotation source present).

## What has changed and why?

Introduces isEnabled flag tracking the state when we should show annotation sources and enable the filtering

## How has it been tested?

Added test

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Summary by CodeRabbit

* **Bug Fixes**
  * Annotation collections menu now displays only when multiple collections exist.
  * Image grid no longer updates filtering or scroll-restoration when collection selection changes.

* **Tests**
  * Added coverage for collection-menu edge cases and deselection behavior, including zero/one collection scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/lightly-ai/lightly-studio/pull/1255?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: lightly_studio_view/src/lib/components/AnnotationCollectionsMenu/AnnotationCollectionsMenu.svelte [hunk 1]
```diff
@@ -20,8 +20,10 @@
 
     let initialized = $state(false);
 
+    const isEnabled = $derived(items.length > 1);
+
     $effect(() => {
-        if (items.length > 0 && !initialized) {
+        if (isEnabled && !initialized) {
```
[the remark this answers is not in the document]
Comment (ikondrat) [reply lightly_studio_view/src/lib/components/AnnotationCollectionsMenu/AnnotationCollectionsMenu.svelte:26]: For different collection we would have reload because it comes from url so the problem will not happen

**Review by mihnea-necsulescu — APPROVED** [review]
